File: src/main/java/com/team3/gudit/auth/service/AuthService.java
package com.team3.gudit.auth.service;

import com.team3.gudit.auth.domain.repository.RefreshTokenRepository;
import com.team3.gudit.auth.redis.RefreshTokenCacheRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class AuthService {
    private final RefreshTokenRepository refreshTokenRepository;
    private final RefreshTokenCacheRepository refreshTokenCacheRepository;

    @Transactional
    public void logout(Long userId) {
        refreshTokenRepository.deleteByUserId(userId);
        refreshTokenCacheRepository.delete(userId);
    }
}

File: src/main/kotlin/com/team3/gudit/auth/service/AuthService.kt
package com.team3.gudit.auth.service

import com.team3.gudit.auth.domain.repository.RefreshTokenRepository
import com.team3.gudit.auth.redis.RefreshTokenCacheRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class AuthService(
    private val refreshTokenRepository: RefreshTokenRepository,
    private val refreshTokenCacheRepository: RefreshTokenCacheRepository,
) {
    @Transactional
    fun logout(userId: Long) {
        refreshTokenRepository.deleteByUserId(userId)
        refreshTokenCacheRepository.delete(userId)
    }
}
